atlasexec/atlas.go

```diff
@@ -300,7 +306,35 @@ func (c *Client) MigrateLint(ctx context.Context, params *MigrateLintParams) (*S
 		args = append(args, "--latest", strconv.FormatUint(params.Latest, 10))
 	}
 	args = append(args, params.Vars.AsArgs()...)
-	return jsonDecode[SummaryReport](c.runCommand(ctx, args, validJSON))
+	return args
+}
+
+// MigrateLint runs the 'migrate lint' command.
+func (c *Client) MigrateLint(ctx context.Context, params *MigrateLintParams) (*SummaryReport, error) {
+	if params.Writer != nil || params.Web {
+		return nil, errors.New("custom Writer or Web reporting are not supported with MigrateLint, use MigrateLintError")
+	}
+	newParams := *params
+	var buf bytes.Buffer
+	newParams.Writer = &buf
+	err := c.MigrateLintError(ctx, &newParams)
+	if err != nil {
+		return nil, err
+	}
+	output := buf.Bytes()
+	err = validJSON(output)
+	return jsonDecode[SummaryReport](bytes.NewReader(output), err)
```

Comment:
I really don't like how much boilerplate this contains
it's also validating the json twice

I wanted to keep it backwards compatible, the original code was using `validJSON`